### PR TITLE
fix bug #42936 - win_iis container_setting

### PR DESCRIPTION
**Pull request by Mapel88**
Module fix
- Failure / success validation is now working properly
- Server 2008 user can now use either numeric or string id

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/win_iis.py [hunk 1]
```diff
@@ -1255,6 +1255,54 @@ def set_container_setting(name, container, settings):
         salt '*' win_iis.set_container_setting name='MyTestPool' container='AppPools'
             settings="{'managedPipeLineMode': 'Integrated'}"
     '''
+
+    def identity_type_map(identity_id,map_type):
```
Comment (twangboy) [inline salt/modules/win_iis.py]: Please put spaces between parameters: `identity, map_type`

File: salt/modules/win_iis.py [hunk 2]
```diff
@@ -1255,6 +1255,54 @@ def set_container_setting(name, container, settings):
...
+        :return: the equivalent id number ot string
+         mapping:
+                'LocalSystem' <--> '0'
+                'LocalService' <--> '1'
+                'NetworkService' <--> '2'
+                'SpecificUser' <--> '3'
+                'ApplicationPoolIdentity' <--> '4'
+        '''
+
+        available_ids = ('0','1','2','3','4','LocalSystem','LocalSystem','NetworkService','SpecificUser','ApplicationPoolIdentity')
```
Comment (twangboy) [inline salt/modules/win_iis.py]: Please place spaces between items in the list.
Second string item should be `LocalService`
Also, wrap after 80 characters:
```
        available_ids = ('0', '1', '2', '3', '4', 'LocalSystem', 'LocalService',
                         'NetworkService', 'SpecificUser',
                         'ApplicationPoolIdentity')
```

File: salt/modules/win_iis.py [hunk 3]
```diff
@@ -1255,6 +1255,54 @@ def set_container_setting(name, container, settings):
...
+                    return 'NetworkService'
+                elif ID == '3':
+                    return 'SpecificUser'
+                elif ID == '4':
+                    return 'ApplicationPoolIdentity'
+                else:
+                    return ID
+        else:
+            raise CommandExecutionError('no such identity type {0}'.format(identity_id))
+
```
Comment (twangboy) [inline salt/modules/win_iis.py]: Actually, maybe using a dictionary would be a little cleaner and remove the need for the `map_type` paramter:

```
        available_ids = {'0': 'LocalSystem',
                         '1': 'LocalService',
                         '2': 'NetworkService',
                         '3': 'SpecificUser',
                         '4': 'ApplicationPoolIdentity',
                         'localsystem': '0',
                         'localservice': '1',
                         'networkservice': '2',
                         'specificuser': '3',
                         'applicationpoolidentity': '4'}
        if ID.lower() in available_ids:
            return available_ids(ID.lower())
        else:
            raise CommandExecutionError('Identity not found: {0}'.format(identity_id))
```

File: salt/modules/win_iis.py [hunk 4]
```diff
@@ -1300,6 +1348,10 @@ def set_container_setting(name, container, settings):
     failed_settings = dict()
 
     for setting in settings:
+        # map identity type from numeric to string for comparing
+        if (container == 'AppPools' and setting == 'processModel.identityType'):
+            settings[setting] = identity_type_map(settings[setting],'string')
+
```
Comment (twangboy) [inline salt/modules/win_iis.py]: If you go with the dictionary approach above, then you wouldn't need the `'string'` parameter.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/win_iis.py (hunk 3 above)
Comment (Mapel88) [inline salt/modules/win_iis.py]: You are correct.
Using dictionary makes much more sense.
Will fix state accordingly

**Pull request closed without merging**